tests/components/tedee/test_binary_sensor.py

```diff
@@ -32,3 +37,27 @@ async def test_binary_sensors(
         entry = entity_registry.async_get(state.entity_id)
         assert entry
         assert entry == snapshot(name=f"entry-{key}")
+
+
+async def test_new_binary_sensors(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    freezer: FrozenDateTimeFactory,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Ensure binary sensors for new lock are added automatically."""
+
+    for key in BINARY_SENSORS:
+        state = hass.states.get(f"binary_sensor.lock_4e5f_{key}")
+        assert state is None
+
+    mock_tedee.locks_dict[666666] = TedeeLock("Lock-4E5F", 666666, 2)
+
+    freezer.tick(timedelta(minutes=10))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    for key in BINARY_SENSORS:
+        state = hass.states.get(f"binary_sensor.lock_4e5f_{key}")
+        assert state
+        assert state == snapshot(name=f"state-{key}")
```

Comment:
I think we should only assert if the state is there, not to completely snapshot it